services/src/test/java/org/keycloak/protocol/oid4vc/issuance/TimeClaimNormalizerTest.java

```diff
@@ -0,0 +1,115 @@
+        Instant orig = Instant.parse("2025-01-02T23:59:59Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.ROUND, 0L, TimeClaimNormalizer.RoundUnit.DAY);
+        Instant normalized = n.normalize(orig);
+        assertThat(normalized, is(Instant.parse("2025-01-02T00:00:00Z")));
+    }
+
+    @Test
+    public void roundHour_truncatesToHour() {
+        Instant orig = Instant.parse("2025-01-02T03:59:59Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.ROUND, 0L, TimeClaimNormalizer.RoundUnit.HOUR);
+        Instant normalized = n.normalize(orig);
+        assertThat(normalized, is(Instant.parse("2025-01-02T03:00:00Z")));
+    }
+
+    @Test
+    public void roundMinute_truncatesToMinute() {
+        Instant orig = Instant.parse("2025-01-02T03:04:59Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.ROUND, 0L, TimeClaimNormalizer.RoundUnit.MINUTE);
+        Instant normalized = n.normalize(orig);
+        assertThat(normalized, is(Instant.parse("2025-01-02T03:04:00Z")));
+    }
+
+    @Test
+    public void roundSecond_truncatesToSecond() {
+        Instant orig = Instant.parse("2025-01-02T03:04:05.987654Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.ROUND, 0L, TimeClaimNormalizer.RoundUnit.SECOND);
+        Instant normalized = n.normalize(orig);
+        assertThat(normalized, is(Instant.parse("2025-01-02T03:04:05Z")));
+    }
+
+    @Test
+    public void defaultRoundUnit_isSecond() {
+        Instant orig = Instant.parse("2025-01-02T03:04:05.987654Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.ROUND, null, null);
+        Instant normalized = n.normalize(orig);
+        assertThat(normalized, is(Instant.parse("2025-01-02T03:04:05Z")));
+    }
+
+    @Test
+    public void randomize_withinWindow_doesNotShiftIntoFuture() {
+        Instant orig = Instant.parse("2025-01-02T22:00:00Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.RANDOMIZE, 3600L, TimeClaimNormalizer.RoundUnit.DAY);
+
+        Instant normalized = n.normalize(orig);
+
+        assertFalse("Normalized time should not be after original time", normalized.isAfter(orig));
+    }
+
+    @Test
+    public void randomize_withinWindow_alwaysBeforeOrEqualOriginal() {
+        Instant orig = Instant.parse("2025-01-02T22:00:00Z");
+        TimeClaimNormalizer n = new TimeClaimNormalizer(TimeClaimNormalizer.Strategy.RANDOMIZE, 3600L, TimeClaimNormalizer.RoundUnit.DAY);
+        Instant normalized = n.normalize(orig);
+        assertFalse("Normalized time should not be after original time", normalized.isAfter(orig));
+        // should be within the randomization window
+        long diffSeconds = orig.getEpochSecond() - normalized.getEpochSecond();
+        assertFalse("Randomized time should be within window", diffSeconds > 3600);
```

Comment:
In my opinion, the following might be good for the readability. But it is up to you to do that or not.
```
        assertTrue("Randomized time should be within window", diffSeconds <= 3600);
```